### PR TITLE
[#5] Implemented Android navigation UI with sample a sample page.

### LINKED ISSUE #5: Setup Android navigation view pages
- navigation view layout implemented

### PR DESCRIPTION
**Pull request by harry2636**
Fixes #5 
- Navigation view UI is added
- Scrolling for webtoon list on the left and preview on the right

### CODE REVIEW

**Comment by dansuh17** [comment]
Please make the two commits into two separate pull requests.

File: .gitignore [hunk 1]
```diff
@@ -1,46 +1 @@
...
-
-# Keystore files
-*.jks
-
-# OSX
-.DS_Store
-
-# etc
-!gradle-wrapper.jar
+#Moved the android gitignore to the Android/toondra.
```
Comment (dansuh17) [inline .gitignore]: This part unnecessary. Please delete.
You can choose to leave needed options such as .DS_Store

**Comment by dansuh17** [comment]
Also, you can add [#issue_num] at the title of the request, so that it is easier to access the issue in the future.

File: Android/toonDra/.gitignore [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+.idea/
+
+# Keystore files
+*.jks
+
+# OSX
+.DS_Store
+
+# etc
+!gradle-wrapper.jar
```
Comment (dansuh17) [inline Android/toonDra/.gitignore:46]: This file is not part of the logical unit of what you have implemented. Please make this a separate pull request.

File: Android/toonDra/app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -3,9 +3,18 @@
...
+    <style name="AppTheme.NoActionBar">
+        <item name="windowActionBar">false</item>
+        <item name="windowNoTitle">true</item>
+    </style>
+
+    <style name="AppTheme.AppBarOverlay" parent="ThemeOverlay.AppCompat.Dark.ActionBar" />
+
+    <style name="AppTheme.PopupOverlay" parent="ThemeOverlay.AppCompat.Light" />
+
+</resources>
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/res/values/styles.xml]: add line

File: Android/toonDra/build.gradle [hunk 4]
```diff
@@ -4,7 +4,7 @@ buildscript {
     jcenter()
   }
   dependencies {
-    classpath 'com.android.tools.build:gradle:2.2.0-beta3'
+    classpath 'com.android.tools.build:gradle:2.2.0'
 
     // NOTE: Do not place your application dependencies here; they belong
     // in the individual module build.gradle files
```
Comment (dansuh17) [inline Android/toonDra/build.gradle:10]: This file change is also not part of your logical unit. Please make this a separate pull request.

File: Android/toonDra/app/src/main/res/layout/app_bar_main.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    <android.support.design.widget.FloatingActionButton
+        android:id="@+id/fab"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="bottom|end"
+        android:layout_margin="@dimen/fab_margin"
+        android:src="@android:drawable/ic_dialog_email" />
+
+</android.support.design.widget.CoordinatorLayout>
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/res/layout/app_bar_main.xml]: Add line

File: Android/toonDra/app/src/main/res/layout/content_main.xml [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+                        android:id="@+id/imageView5"
+                        android:background="#BBBBBB"
+                        android:src="@drawable/fd_3"
+                        android:adjustViewBounds="true"/>
+
+                </LinearLayout>
+            </ScrollView>
+        </LinearLayout>
+    </LinearLayout>
+</RelativeLayout>
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/res/layout/content_main.xml]: add line

File: Android/toonDra/app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,16 +1,26 @@
...
-        android:text="Hello World!" />
-</RelativeLayout>
+        android:layout_height="match_parent"
+        android:layout_gravity="start"
+        android:fitsSystemWindows="true"
+        app:headerLayout="@layout/nav_header_main"
+        app:menu="@menu/activity_main_drawer" />
+
+</android.support.v4.widget.DrawerLayout>
+
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/res/layout/activity_main.xml]: unnecessary blank line

File: Android/toonDra/app/src/main/res/drawable/side_nav_bar.xml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<shape xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shape="rectangle">
+    <gradient
+        android:angle="135"
+        android:centerColor="#4CAF50"
+        android:endColor="#2E7D32"
+        android:startColor="#81C784"
+        android:type="linear" />
+</shape>
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/res/drawable/side_nav_bar.xml]: add line

File: Android/toonDra/app/src/main/res/drawable/ic_menu_manage.xml [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportHeight="24.0"
+    android:viewportWidth="24.0">
+    <path
+        android:fillColor="#FF000000"
+        android:pathData="M22.7,19l-9.1,-9.1c0.9,-2.3 0.4,-5 -1.5,-6.9 -2,-2 -5,-2.4 -7.4,-1.3L9,6 6,9 1.6,4.7C0.4,7.1 0.9,10.1 2.9,12.1c1.9,1.9 4.6,2.4 6.9,1.5l9.1,9.1c0.4,0.4 1,0.4 1.4,0l2.3,-2.3c0.5,-0.4 0.5,-1.1 0.1,-1.4z" />
+</vector>
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/res/drawable/ic_menu_manage.xml]: add line

File: Android/toonDra/app/build.gradle [hunk 10]
```diff
@@ -28,4 +28,5 @@ dependencies {
   compile fileTree(include: ['*.jar'], dir: 'libs')
   compile 'com.android.support:appcompat-v7:24.2.1'
   testCompile 'junit:junit:4.12'
+  compile 'com.android.support:design:24.2.1'
```
Comment (dansuh17) [inline Android/toonDra/app/build.gradle:31]: place this compile dependency line grouped along with other compile dependencies - above the testCompile dependency group

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java [hunk 11]
```diff
@@ -1,12 +1,119 @@
...
+      @Override
+      public void onClick(View view) {
+        Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
+                .setAction("Action", null).show();
+      }
+    });
+
+    DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+    ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+            this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: line longer than 100.
split the arguments into separate lines

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java [hunk 12]
```diff
@@ -1,12 +1,119 @@
...
+    navigationView.setNavigationItemSelectedListener(this);
+
+    //test code
+    LinearLayout linearLayout = (LinearLayout) findViewById(R.id.linearLayout1);
+    linearLayout.setOnClickListener(mClickListener);
+  }
+  LinearLayout.OnClickListener mClickListener = new View.OnClickListener() {
+    public void onClick(View v) {
+      TextView textFruit = (TextView) findViewById(R.id.title_text1);
+      if((flag++)%2 == 1) {
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: a space after 'if'.
also a space around % operator
`if ((flag++) % 2 == 1) {`
...

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java [hunk 13]
```diff
@@ -1,12 +1,119 @@
...
+        textFruit.setText("changed");
+      }
+      else {
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: place else statement at the same line with ending braces of if statement.
`} else {`

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java [hunk 14]
```diff
@@ -1,12 +1,119 @@
...
+
+    NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+    navigationView.setNavigationItemSelectedListener(this);
+
+    //test code
+    LinearLayout linearLayout = (LinearLayout) findViewById(R.id.linearLayout1);
+    linearLayout.setOnClickListener(mClickListener);
+  }
+  LinearLayout.OnClickListener mClickListener = new View.OnClickListener() {
+    public void onClick(View v) {
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java:52]: inappropriate parameter name - should not be a single character
`view` recommended

**Review by dansuh17 — COMMENTED** [review]
You can run `./gradlew checkstyle` to avoid style errors. Please run the command before making a pull request.

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java [hunk 15]
```diff
@@ -1,12 +1,119 @@
...
+      public void onClick(View view) {
+        Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
+                .setAction("Action", null).show();
+      }
+    });
+
+    DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+    ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+            this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
+    drawer.setDrawerListener(toggle);
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: I get 'deprecated' message on setDrawerListener. Perhaps change this to `addDrawerListener()`?

[see link](http://stackoverflow.com/questions/35639454/method-setdrawerlistener-is-deprecated)

File: Android/toonDra/.gitignore (hunk 2 above)
Comment (harry2636) [reply Android/toonDra/.gitignore:46]: Reverted the corresponding commit.

File: Android/toonDra/app/build.gradle (hunk 10 above)
Comment (harry2636) [reply Android/toonDra/app/build.gradle:31]: I first grouped it along with others but Android Studio automatically replaced it like above.

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java (hunk 15 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Changed it to addDrawerListener.

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java (hunk 12 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Changed it.

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java (hunk 13 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Changed it.

File: Android/toonDra/app/src/main/res/drawable/ic_menu_manage.xml (hunk 9 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/res/drawable/ic_menu_manage.xml]: Added.

File: Android/toonDra/app/src/main/res/drawable/side_nav_bar.xml (hunk 8 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/res/drawable/side_nav_bar.xml]: Added

File: Android/toonDra/app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/res/layout/activity_main.xml]: Deleted.

File: Android/toonDra/app/src/main/res/layout/app_bar_main.xml (hunk 5 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/res/layout/app_bar_main.xml]: Added.

File: Android/toonDra/app/src/main/res/layout/content_main.xml (hunk 6 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/res/layout/content_main.xml]: Added.

File: Android/toonDra/app/src/main/res/values/styles.xml (hunk 3 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/res/values/styles.xml]: Added.

File: Android/toonDra/build.gradle (hunk 4 above)
Comment (harry2636) [reply Android/toonDra/build.gradle:10]: Changed it.

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java (hunk 14 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java:52]: It's android recommended practice because this parameter is never used in the onClick() function.(It is used to auto-referenced by findViewbyId() function.)

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java (hunk 11 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Changed it.

**Comment by harry2636** [comment]
@deNsuh changed according to the requests.

File: .gitignore (hunk 1 above)
Comment (harry2636) [reply .gitignore]: Changed it and moved it to new pull request #16

File: Android/toonDra/.gitignore (hunk 2 above)
Comment (dansuh17) [reply Android/toonDra/.gitignore:46]: nope the changes are made with commits stacked.
this will give conflict when the other pull request is made.
use `git commit --amend` and `git push -f`

**Review by dansuh17 — CHANGES_REQUESTED** [review]

**Comment by dansuh17** [comment]
Again, a pull request should be a **single** commit.
use `git rebase -i HEAD~4`

**Comment by harry2636** [comment]
@deNsuh Reflected the requests. Please check it. And thanks for giving nice feedbacks with appropriate links!

**Review by dansuh17 — APPROVED** [review]

**Comment by deNsuh** [comment]
LGTM 👍

File: Android/toonDra/.gitignore (hunk 2 above)
Comment (dansuh17) [inline Android/toonDra/.gitignore:46]: This has been already done in commit 0f3915b.
Remove changes for this file (you can choose not to include this file for commit).
Also it is a good practice to always submit a pull request with your commit rebased to latest master commit.
`git rebase master`

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java [hunk 16]
```diff
@@ -1,12 +1,119 @@
...
+    ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
+            this, drawer, toolbar, R.string.navigation_drawer_open,
+            R.string.navigation_drawer_close);
+    drawer.addDrawerListener(toggle);
+    toggle.syncState();
+
+    NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+    navigationView.setNavigationItemSelectedListener(this);
+
+    //test code
```
Comment (dansuh17) [inline Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: what do you mean by 'test code'? Is this part of the code not included for production?

**Review by dansuh17 — CHANGES_REQUESTED** [review]

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java (hunk 16 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Nope it would be in the production, I forgot to erase the comment.

File: Android/toonDra/.gitignore (hunk 2 above)
Comment (harry2636) [reply Android/toonDra/.gitignore:46]: Thanks for tip, I changed it.

**Comment by harry2636** [comment]
@deNsuh @heewoonam Reflected the requests.

**Review by dansuh17 — APPROVED** [review]

**Comment by dansuh17** [comment]
LGTM

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java [hunk 17]
```diff
@@ -1,12 +1,118 @@
...
+    drawer.addDrawerListener(toggle);
+    toggle.syncState();
+
+    NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
+    navigationView.setNavigationItemSelectedListener(this);
+
+    LinearLayout linearLayout = (LinearLayout) findViewById(R.id.linearLayout1);
+    linearLayout.setOnClickListener(mClickListener);
+  }
+  LinearLayout.OnClickListener mClickListener = new View.OnClickListener() {
```
Comment (heewoonam) [inline Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Change OnClickListener to private final

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java [hunk 18]
```diff
@@ -1,12 +1,118 @@
...
+    public void onClick(View v) {
+      TextView textFruit = (TextView) findViewById(R.id.title_text1);
+      if ((flag++) % 2 == 1) {
+        textFruit.setText("changed");
+      } else {
+        textFruit.setText("returned");
+      }
+    }
+  };
+  int flag = 1;
```
Comment (heewoonam) [inline Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Change to private

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java (hunk 17 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Changed it.

File: Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java (hunk 18 above)
Comment (harry2636) [reply Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/MainActivity.java]: Changed it.

**Comment by harry2636** [comment]
@deNsuh @heewoonam Fixed OOM bugs and reflected the requests.

**Comment by heewoonam** [comment]
LGTM

**Comment by dansuh17** [comment]
LGTM
